modules/lib/src/main/resources/assets/js/v6/features/store/contentTreeUpload.store.ts

```diff
@@ -0,0 +1,24 @@
+import {map} from 'nanostores';
+import {ContentUploadData} from '../views/browse/grid/ContentUploadData';
+
+type ContentTreeUploadStore = Record<string, ContentUploadData>;
+
+export const $contentTreeUpload = map<ContentTreeUploadStore>({});
+
+export function addContentTreeUploadItem(id: string, name: string, parentId = ''): void {
```

Comment:
Marking `parentId` optional might be a better choice if you want to have "unset" option, that will fall back to root by default.